source/utils/network.ts

```diff
@@ -70,7 +70,13 @@ export const verifyNetworkEIP1559Compatibility = async (
 ) => {
   try {
     const latestBlock = await web3Provider.getBlock('latest');
```

Comment:
check if using the block in the state here can potentially reduce 1 API call `const currentBlockNumber = store.getState().vault.currentBlock?.number;` if there's no block in the state then `await web3Provider.getBlock('latest');` or use just the API call if the state block doesn't work for this case.